elastic/integrations
# [ess_billing] Addition of an ESS billing integration

## Proposed commit message

This adds the `ess_billing` integration package, that consumes the ESS billing API at https://api.elastic-cloud.com. This in turns gives a richer view than what is directly exposed through the cloud organization page, and allows customers to gather clearer insights in their spending patterns.

Currently, a single `billing` dataset is implemented, performing daily CEL API calls against the API. Lookbehind is supported, with 1 year configured as default (meaning initial run of the input will fetch the last year of billing data).

## Checklist


I don't have access to https://docs.elastic.dev - not quite sure how to check the last bullet point.

## Author's Checklist



## How to test this PR locally

You will need an active cloud organization, for which you want to grab:
* The organization ID, visible on the organization page.
* An API key with billing permissions.

Simply install the integration without changing the defaults, and you should see the included dashboard getting populated with data. Make sure you change your time range, as the default will initially populate data from 1 year in the past.

## Related issues

N/A

## Screenshots

<img width="1678" alt="image" src="https://github.com/user-attachments/assets/50147537-c02b-45c6-936a-b0aec796b5d5">

![ess_billing-dashboard](https://github.com/user-attachments/assets/bad96d25-b0ce-49a2-b19e-f8bb7a1be0d0)

The cloud locations will need fine tuning, but I can figure this out while we keep the integration in beta.
I've also taken some liberties with regards to ECS, as we don't really have good handling for such information AFAICT.
